library/x509_create.c

```diff
@@ -143,6 +149,9 @@ int mbedtls_x509_string_to_names( mbedtls_asn1_named_data **head, const char *na
 
         if( !in_tag && s != c + 1 )
         {
+            if( *c == '\"' )
```

Comment:
I do not think this works well in some cases. If I understand correctly, you are trying to match pairs of `"` characters so that `,` are note interpreted as separators. However, this does not consider a distinguished name of the form:
```
C=NL,O=Of\"fspark\, Inc., OU=PolarSSL
```
In this case the`"` should be interpreted as part of the organisation instead of the start of a quoted string.